jenkins/sparklingWaterPipeline.groovy

```diff
@@ -70,6 +82,33 @@ def prepareSparklingWaterEnvironment() {
     return { config ->
         stage('QA: Prepare Sparkling Water Environment') {
 
+            // In case of nightly build, modify gradle.properties
+            if(config.buildNightly.toBoolean()){
+                def h2oNightlyBuildVersion = sh(
+                        script: "curl http://h2o-release.s3.amazonaws.com/h2o/master/latest",
+                        returnStdout: true
+                ).trim()
+
+                def h2oNightlyMajorVersion = sh(
+                        script: "curl http://h2o-release.s3.amazonaws.com/h2o/master/${h2oNightlyBuildVersion}/project_version | cut -f1-3 -d\".\"",
+                        returnStdout: true
+                ).trim()
+
+                sh  """
```

Comment:
Minor note: you do not need shell to process properties in this sense, maybe it could be easier to generate new one...or read them with help of groovy.

Reply:
good point, I can use groovy to read the content of the urls